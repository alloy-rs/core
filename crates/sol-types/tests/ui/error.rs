use ethers_sol_types::sol;

sol! {
    error MissingParens1
}

sol! {
    error MissingParens2;
}

sol! {
    error MissingSemi()
}

sol! {
    error TrailingComma(uint256,);
    error Valid(uint256);
}

fn main() {}
