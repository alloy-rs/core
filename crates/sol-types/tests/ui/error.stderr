error: unexpected end of input, expected parentheses
 --> tests/ui/error.rs:3:1
  |
3 | / sol! {
4 | |     error MissingParens1
5 | | }
  | |_^
  |
  = note: this error originates in the macro `sol` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected parentheses
 --> tests/ui/error.rs:8:25
  |
8 |     error MissingParens2;
  |                         ^
  |
  = note: this error originates in the macro `sol` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `;`
  --> tests/ui/error.rs:11:1
   |
11 | / sol! {
12 | |     error MissingSemi()
13 | | }
   | |_^
   |
   = note: this error originates in the macro `sol` (in Nightly builds, run with -Z macro-backtrace for more info)
